pages/apply/submit/declaration.tsx

```diff
@@ -27,72 +31,108 @@ const Declaration = (): JSX.Element => {
   const mainResident = useAppSelector((s) => s.application.mainApplicant);
   const application = useAppSelector((store) => store.application);
 
+  const [loading, setLoading] = useState<boolean>(false);
+  const [userError, setUserError] = useState<string | null>(null);
+
   const submitApplication = async () => {
     const [isEligible, reasons] = checkEligible(application);
+    setUserError(null);
+    setLoading(true);
     if (!isEligible) {
       const reasonStrings = reasons.map((reason) =>
         getDisqualificationReasonOption(reason)
       );
       const reason = reasonStrings.join(',');
+      // this is just a notification, so for now we don't need to handle errors
       dispatch(sendDisqualifyEmail({ application, reason }));
-      dispatch(disqualifyApplication(application.id!));
-      router.push('/apply/not-eligible');
+
+      try {
+        console.log('hello');
+        await dispatch(disqualifyApplication(application.id!)).unwrap();
+        router.push('/apply/not-eligible');
+      } catch (error) {
+        setUserError(error as string);
```

Comment:
should `setLoading(false);` be here too?

Reply:
the finally should set the loading state back to false in the try or catch